content/chainguard/chainguard-enforce/chainguard-enforce-kubernetes/gulfstream-overview/index.md

```diff
@@ -0,0 +1,90 @@
+
+One strategy used to isolate workloads is to run them in separate node pools. This way, even with a container escape, the workloads are running on different nodes. However, if the attacker gains access toAPI server credentials, they can potentially move laterally and try to compromise workloads on different nodes.
+
+So without hard multi-tenant isolation between a control plane and user workloads, can a control plane truly be considered “trusted”? In the event that a workload is compromised, you can't be certain that an attacker cannot move laterally into the trusted control plane process.
+
+
+## Where Gulfstream comes in
+
+As mentioned in the introduction, Gulfstream is a streaming controller architecture for Kubernetes. It was developed to combat the cost of control plane extensions at scale, make costs proportional to usage, and amortize the resource usage over the clusters being extended.
+
+Gulfstream encapsulates the Controller and Webhook responsibilities into CRDs. It also has a *Meta Control Plane* (MCP) that understands how to handle these CRDs and delegate the Reconciler business logic to SaaS.
+
+Gulfstream has two main modes of execution. Firstly, it supports users running their own MCP, also known as the "Chainguard Enforce Agent," which is useful for private clusters, or clusters behind a firewall. Chainguard also offers a fully SaaS Gulfstream, sometimes referred to as "agentless." In this case, the MCP runs on a cluster managed by Chainguard, but is pointed at a customer's cluster to work much like “agentful”. This enables a hard multi-tenant separation between our trusted control plane and the workloads running on your cluster.
+
+Chainguard's control plane doesn’t run alongside user code, so it can provide certain security assurances that are much harder to achieve when running on the same cluster. Chainguard Enforce speaks directly to your Kubernetes API server using your cloud provider’s IAM to monitor your cluster’s workloads, and your cluster’s API server talks directly to Chainguard Enforce for admission control decisions based on policy. So even if a workload on your cluster were compromised, it is no easier for the attacker to compromise a Chainguard trusted control plane than attacking us head on.
+
+
+## Learn More
+
+As mentioned previously, running Chaingaurd Enforce in its agentless mode allows for hard multi-tenant separation between Chaingaurd's trusted control plane and your workloads. You can get started with agentless connections by first [setting up a cloud account association](../cloud-account-associations/) and then you can [use Discovery to discover your cloud resources](../chainguard-enforce-discovery-onboarding/) and then install Chainguard Enforce in Agentless mode.
```

Comment:
```suggestion
As mentioned previously, running Chainguard Enforce in its agentless mode allows for hard multi-tenant separation between Chainguard's trusted control plane and your workloads. You can get started with agentless connections by first [setting up a cloud account association](../cloud-account-associations/), then you can [use Discovery to discover your cloud resources](../chainguard-enforce-discovery-onboarding/), and finally install Chainguard Enforce in Agentless mode.
```

Reply:
of course the last paragraph I wrote is riddled with errors